fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/metadata/SchemaTemplateSerDeTests.java

```diff
@@ -433,6 +433,59 @@ public void onlyQueriedSqlFunctionsAreCompiled() throws RelationalException {
         Assertions.assertTrue(peekingDeserializer.hasOneCompilationRequestFor("SqlFunction3"));
     }
 
+    @Test
+    public void schemaTemplateToBuilderPreservesIntermingledTablesFlag() {
+        var sampleRecordSchemaTemplate = RecordLayerSchemaTemplate.newBuilder()
+                .setName("TestSchemaTemplate")
+                .setVersion(42)
+                .addAuxiliaryType(DataType.StructType.from(
+                        "Subtype",
+                        List.of(DataType.StructType.Field.from("field1", DataType.Primitives.INTEGER.type(), 0)),
+                        true))
+                .setIntermingleTables(true)
+                .addTable(
+                        RecordLayerTable.newBuilder(true)
+                                .setName("T1")
+                                .addColumn(RecordLayerColumn.newBuilder()
+                                        .setName("COL1")
+                                        .setDataType(
+                                                DataType.StructType.from(
+                                                        "Subtype",
+                                                        List.of(DataType.StructType.Field.from("field1", DataType.Primitives.INTEGER.type(), 1)),
+                                                        true))
+                                        .build())
+                                .build())
+                .build();
+
+        var newBuilder = sampleRecordSchemaTemplate.toBuilder();
+        Assertions.assertTrue(newBuilder.isIntermingleTables());
+
+        sampleRecordSchemaTemplate = RecordLayerSchemaTemplate.newBuilder()
+                .setName("TestSchemaTemplate")
+                .setVersion(42)
+                .addAuxiliaryType(DataType.StructType.from(
+                        "Subtype",
+                        List.of(DataType.StructType.Field.from("field1", DataType.Primitives.INTEGER.type(), 0)),
+                        true))
+                .setIntermingleTables(false)
```

Comment:
These two cases are the same except for how they set intermingle tables, right? This could be a parameterized test where the value of `setIntermingleTables` is a parameter to the test, which would help to avoid some code repetition here